Tests/RPCResponseHandlerTest.swift

```diff
@@ -0,0 +1,181 @@
+    XCTAssertEqual(tezosKitError.underlyingError, testErrorString)
+  }
+
+  public func testHandleResponseWithHTTPInvalid() {
+    let response = httpResponse(with: 9_000)
+    let (result, error) = responseHandler.handleResponse(
+      response: response,
+      data: testErrorStringData,
+      error: TestError.testError,
+      responseAdapterClass: StringResponseAdapter.self
+    )
+
+    XCTAssertNil(result)
+    XCTAssertNotNil(error)
+
+    guard let tezosKitError = error as? TezosClientError else {
+      XCTFail()
+      return
+    }
+
+    XCTAssertEqual(tezosKitError.kind, .unknown)
+    XCTAssertEqual(tezosKitError.underlyingError, testErrorString)
+  }
+
+  public func testHandleResponseWithHTTP200AndError() {
+    let response = httpResponse(with: 200)
+    let testError = TestError.testError
+    let (result, error) = responseHandler.handleResponse(
+      response: response,
+      data: testParsedStringData,
+      error: testError,
+      responseAdapterClass: StringResponseAdapter.self
+    )
+
+    XCTAssertNotNil(error)
+    XCTAssertNil(result)
+
+    guard let tezosKitError = error as? TezosClientError else {
+      XCTFail()
+      return
+    }
+
+    XCTAssertEqual(tezosKitError.kind, .rpcError)
+    XCTAssertEqual(tezosKitError.underlyingError, testError.localizedDescription)
+  }
+
+  public func testHandleResponseWithHTTP200AndNilErrorAndValidData() {
+    let response = httpResponse(with: 200)
+    let (result, error) = responseHandler.handleResponse(
+      response: response,
+      data: testParsedStringData,
+      error: nil,
+      responseAdapterClass: StringResponseAdapter.self
+    )
+
+    XCTAssertNil(error)
+    XCTAssertNotNil(result)
+    XCTAssertEqual(result, testParsedString)
+  }
+
+  public func testHandleResponseWithHTTP200AndNilErrorAndNoData() {
+    let response = httpResponse(with: 200)
+    let (result, error) = responseHandler.handleResponse(
+      response: response,
+      data: nil,
+      error: nil,
+      responseAdapterClass: StringResponseAdapter.self
+    )
+
+    XCTAssertNotNil(error)
+    XCTAssertNil(result)
+
+    guard let tezosKitError = error as? TezosClientError else {
+      XCTFail()
+      return
+    }
+
+    XCTAssertEqual(tezosKitError.kind, .unexpectedResponse)
+    XCTAssertEqual(tezosKitError.underlyingError, nil)
+  }
+
+  public func testHandleResponseWithHTTP200AndNilErrorAndBadData() {
+    let response = httpResponse(with: 200)
+    let (result, error) = responseHandler.handleResponse(
+      response: response,
+      data: testParsedStringData,
+      error: nil,
+      responseAdapterClass: PeriodKindResponseAdapter.self
+    )
+
+    XCTAssertNotNil(error)
+    XCTAssertNil(result)
+
+    guard let tezosKitError = error as? TezosClientError else {
+      XCTFail()
+      return
+    }
+
+    XCTAssertEqual(tezosKitError.kind, .unexpectedResponse)
+    XCTAssertEqual(tezosKitError.underlyingError, nil)
+  }
+
```

Comment:
// MARK: - Helpers

or something split the two groups

Reply:
Done